sklearn/ensemble/_forest.py

```diff
@@ -324,6 +324,14 @@ def fit(self, X, y, sample_weight=None):
             # [:, np.newaxis] that does not.
             y = np.reshape(y, (-1, 1))
 
+        if self.criterion == "poisson":
+            if np.any(y < 0):
+                raise ValueError("""Some value(s) of y are negative which is \
+                                    not allowed for Poisson regression.""")
+            if np.sum(y) <= 0:
+                raise ValueError("""Sum of y is not positive which is strictly \
+                                    necessary for Poisson regression.""")
```

Comment:
Same here.
Could you put the **strictly** ealier, i.e. `Sum of y is not strictly positive which is necessary...`.

Reply:
Thanks @lorentzenchr ! I'll put these changes through.